blazor/datagrid/paging.md

```diff
@@ -7,443 +7,1162 @@ control: DataGrid
-        Int32.TryParse(this.DefaultGrid.Height, out GridHeight); //datagrid height
-        var PageSize = (this.DefaultGrid.PageSettings as GridPageSettings).PageSize; //initial page size
-        decimal PageResize = ((GridHeight) - (PageSize * RowHeight)) / RowHeight; //new page size is obtained here
-        #pragma warning disable BL0005
-        (this.DefaultGrid.PageSettings as GridPageSettings).PageSize = PageSize + (int)Math.Round(PageResize);
-        #pragma warning restore BL0005
+        if(args.Value != string.Empty)
+        {
+            value = int.Parse(args.Value);
+        }
     }
 }
+{% endhighlight %}
+{% highlight c# tabtitle="OrderData.cs" %}
+ public class OrderData
+    {
+        public static List<OrderData> Orders = new List<OrderData>();
 
-```
+        public OrderData()
+        {
 
-N> You can achieve better performance by using datagrid paging to fetch only a pre-defined number of records from the data source.
+        }
+        public OrderData(int? OrderID,string CustomerID,string ShipCity, string ShipName)
+        {
+            this.OrderID = OrderID;    
+            this.CustomerID = CustomerID;
+            this.ShipCity = ShipCity;
+            this.ShipName = ShipName;           
+        }
 
-## Pager with page size dropdown
+        public static List<OrderData> GetAllRecords()
+        {
+            if (Orders.Count() == 0)
+            {
+               
+                int OrderID = 10248;
+
+                for (int i = 1; i < 7; i++)
+                {
+                    Orders.Add(new OrderData(OrderID+1, "VINET", "Reims", "Vins et alcools Chevali"));
+                    Orders.Add(new OrderData(OrderID+2, "TOMSP", "Münster", "Toms Spezialitäten"));
+                    Orders.Add(new OrderData(OrderID+3, "HANAR", "Rio de Janeiro", "Hanari Carnes"));
+                    Orders.Add(new OrderData(OrderID+4, "VICTE", "Lyon", "Victuailles en stock"));
+                    Orders.Add(new OrderData(OrderID+5, "SUPRD", "Charleroi", "Suprêmes délices"));
+                    Orders.Add(new OrderData(OrderID+6, "HANAR", "Lyon", "Hanari Carnes"));
+                    Orders.Add(new OrderData(OrderID+7, "CHOPS", "Rio de Janeiro", "Chop-suey Chinese"));
+                    Orders.Add(new OrderData(OrderID+8, "RICSU", "Münster", "Richter Supermarkt"));
+                    Orders.Add(new OrderData(OrderID+9, "WELLI", "Reims", "Wellington Import"));
+                   
+                    OrderID += 9;
+                }
+            }
+            return Orders;
+        }
+
+        public int? OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public string ShipCity { get; set; }
+        public string ShipName { get; set; }
+   }
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "https://blazorplayground.syncfusion.com/embed/VjLztiturzdVycbb?appbar=false&editor=false&result=true&errorlist=false&theme=bootstrap5" %}
```

Comment:
change SftextBox to SfNumericTextBox without arrow icons

Reply:
Resolved